packages/webapp/pages/reset-password.tsx

```diff
@@ -1,28 +1,90 @@
-import React, { ReactElement } from 'react';
+import React, { FormEvent, ReactElement, useState } from 'react';
 import DailyCircle from '@dailydotdev/shared/src/components/DailyCircle';
 import Logo from '@dailydotdev/shared/src/components/Logo';
 import { TextField } from '@dailydotdev/shared/src/components/fields/TextField';
 import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import { useQuery, useMutation } from 'react-query';
+import {
+  errorsToJson,
+  getNodeValue,
+  getSettingsFlow,
+  RegistrationParameters,
+  resetPassword,
+} from '@dailydotdev/shared/src/lib/auth';
+import { disabledRefetch } from '@dailydotdev/shared/src/lib/func';
+import { useRouter } from 'next/router';
+import { formToJson } from '@dailydotdev/shared/src/lib/form';
 
 function ResetPassword(): ReactElement {
+  const router = useRouter();
+  const [hint, setHint] = useState(null);
+  const { data } = useQuery(
+    'settings',
+    () => getSettingsFlow(router.query.flow as string),
+    {
+      ...disabledRefetch,
+      retry: false,
+      enabled: !!router.query.flow,
+    },
+  );
+  const { mutateAsync: reset } = useMutation(resetPassword, {
+    onSuccess: ({ error, data: success }) => {
+      if (success) {
+        return window.location.replace(process.env.NEXT_PUBLIC_WEBAPP_URL);
+      }
+
+      if (!error?.ui) {
+        return setHint('Session could have expired!');
+      }
+
+      const json = errorsToJson<keyof RegistrationParameters>(error);
+      return setHint(json?.password);
+    },
+  });
+
+  const onSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const { password } = formToJson<{ password: string }>(e.currentTarget);
+    const params: RegistrationParameters = {
+      method: 'password',
+      csrf_token: getNodeValue('csrf_token', data.ui.nodes),
+      'traits.email': getNodeValue('traits.email', data.ui.nodes),
+      'traits.fullname': getNodeValue('traits.fullname', data.ui.nodes),
+      'traits.username': getNodeValue('traits.username', data.ui.nodes),
+      'traits.image': getNodeValue('traits.image', data.ui.nodes),
```

Comment:
Why does the reset password flow care about all these fields? that seems weird?

Reply:
The account recovery flow would allow you to update your profile details as well, basically, the settings flow, but with privileged access already. So to validate the transaction it requires all the necessary information if some of the fields are required.